File: src/main/java/com/undsf/mcmodmgr/curseforge/models/AnyResponse.java
package com.undsf.mcmodmgr.curseforge.models;

import java.util.LinkedHashMap;
import java.util.Map;

public class AnyResponse extends DataResponse<Map<String, Object>> {
    public AnyResponse() {
    }

    public void put(String key, Object value) {
        if (data == null) {
            data = new LinkedHashMap<>();
        }
        data.put(key, value);
    }

    public Object get(String key, Object defaultValue) {
        if (data == null) {
            data = new LinkedHashMap<>();
        }
        return data.getOrDefault(key, defaultValue);
    }

    public String getString(String key, String defaultValue) {
        Object value = data.getOrDefault(key, defaultValue);
        return (String) value;
    }

    public Integer getInteger(String key, int defaultValue) {
        Object value = data.getOrDefault(key, defaultValue);
        return (Integer) value;
    }
}

File: src/main/kotlin/com/undsf/mcmodmgr/models/AnyResponse.kt
package com.undsf.mcmodmgr.models

class AnyResponse(code: Int, message: String?) : DataResponse<MutableMap<String, Any?>>(code, message) {
    init {
        data = mutableMapOf()
    }

    fun addParam(key: String, value: Any?) {
        data?.put(key, value)
    }
}